integrations/statuspage/client.py

```diff
@@ -0,0 +1,115 @@
+from enum import StrEnum
+from typing import Any, AsyncGenerator, Optional
+from loguru import logger
+from port_ocean.utils import http_async_client
+
+PAGE_SIZE = 50
+
+
+class ResourceKey(StrEnum):
+    COMPONENT_GROUPS = "component-groups"
+    COMPONENTS = "components"
+    INCIDENTS = "incidents"
+
+
+class StatusPageClient:
+    def __init__(
+        self, statuspage_api_key: str, statuspage_ids: Optional[list[str]] = None
+    ) -> None:
+        self.statuspage_url = "https://api.statuspage.io/v1/pages"
+        self.statuspage_api_key = statuspage_api_key
+        self.statuspage_ids = statuspage_ids
+
+        self.client = http_async_client
+        self.api_auth_header = {"Authorization": self.statuspage_api_key}
+        self.client.headers.update(self.api_auth_header)
+
+    async def _get_paginated_resources(
+        self, endpoint: str, params: dict[str, Any] = {}
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        page = 1
+        while True:
+            params["page"] = page
+            response = await self.client.get(
+                f"{self.statuspage_url}/{endpoint}", params=params
+            )
+            response.raise_for_status()
+            data = response.json()
+            if not data:
+                break
+            yield data
+            page += 1
+
+    async def get_pages(self) -> list[dict[str, Any]]:
+        async for pages in self._get_paginated_resources(""):
```

Comment:
Change this to the actual endpoint you are referencing. Change the `self.statuspage_url` variable accordingly.
```suggestion
        async for pages in self._get_paginated_resources("v1/pages"):
```

Reply:
I did this since [every other resource is under pages](https://developer.statuspage.io/#tag/pages).

- `v1/pages/<page-id>/incidents`
- `v1/pages/<page-id>/components`
- `v1/pages/<page-id>/component_groups`

Would you rather write out the same `v1/pages` in all the individual method calls for the different resources

Reply:
Maybe have it stored in a variable and reference this variable. It needs to be explicit.